GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -873,68 +975,68 @@ private FolderData AddFileToTree(GitIndexEntry indexEntry)
         {
             if (!this.projectionFolderCache.TryGetValue(folderPath, out folderData))
             {
-                LazyUTF8String[] pathParts = folderPath
-                    .Split(new char[] { Path.DirectorySeparatorChar }, StringSplitOptions.RemoveEmptyEntries)
-                    .Select(x => new LazyUTF8String(x))
-                    .ToArray();
-
-                FolderEntryData data;
-                if (!this.TryGetFolderEntryDataFromTree(pathParts, folderEntryData: out data))
+                if (!this.TryGetFolderDataFromTreeUsingPath(folderPath, out folderData) ||
+                    !folderData.IsIncluded)
                 {
-                    folderPath = null;
+                    folderData = null;
                     return false;
                 }
 
-                if (data.IsFolder)
-                {
-                    folderData = (FolderData)data;
-                    this.projectionFolderCache.TryAdd(folderPath, folderData);
-                }
-                else
-                {
-                    EventMetadata metadata = CreateEventMetadata();
-                    metadata.Add("folderPath", folderPath);
-                    metadata.Add(TracingConstants.MessageKey.InfoMessage, "Found file at path");
-                    this.context.Tracer.RelatedEvent(
-                        EventLevel.Informational,
-                        $"{nameof(this.TryGetOrAddFolderDataFromCache)}_FileAtPath",
-                        metadata);
-
-                    folderPath = null;
-                    return false;
-                }
+                this.projectionFolderCache.TryAdd(folderPath, folderData);
             }
 
             return true;
         }
 
         /// <summary>
-        /// Finds the FolderEntryData for the path provided
+        /// Takes a path and get the FolderData object fo that path if it exist and is a folder

```

Comment:
Typos: 
`get` -> `gets`
`object fo that` -> `object for that`
`if it exist` -> `if it exists`